### ISSUE TITLE
Grouping different methods or functions

### ISSUE THREAD

**Issue by tshort** [MichaelHatherly/Docile.jl#63]
Based on a post to julia-users to group similar methods, maybe the following would be useful:

```julia
"""
doc string...
"""
[myfun, myfun!]   # or symbol equivalents
```

Maybe it could be extended to also include multiple methods:

```julia
"""
doc string...
"""
[(fill, :MyType, Any...), (fill!, :MyType, Any...)]   
```

**Comment by MichaelHatherly** [comment]
While implementing the basic method groups we currently have I did suspect this one would probably come up. Although allowing something like:

```julia
"""
doc string...
"""
[function_that_does_something, function_that_does_something_completely_different]
```

kind of fills me with dread, I think allowing a general syntax for this is the best way to go.

I'd think people should only use this to document mutating and non-mutating versions of a function. I can't think of a case where associating totally different names with the same `Entry` would be all that necessary or should even be encouraged.

(So basically I'm fine with adding something that does this, but it's proper use should be explained prominently in the docs.)

The `exec` method should be able to handle this kind of syntax fine I'd think.

We could also possibly allow the product of a set of functions and a set of signatures to perhaps reduce repetition a little. Not sure yet about how it should look though.

**Comment by MichaelHatherly** [comment]
Implemented with your suggested sytnax @tshort in 5e192af53129e1a3e52693abc22203b850f945ab. It's limited to either a set of methods or a set of functions; not a mix of the two since that's a limitation of the `Entry{category}` system currently used.
